front/app/containers/Admin/reporting/components/ReportBuilder/TopBar/index.tsx

```diff
@@ -130,39 +127,42 @@ const ContentBuilderTopBar = ({
       return;
     }
 
-    const nodes = query.getSerializedNodes();
-    const firstNode = nodes.ROOT?.nodes[0];
+    const interval = setInterval(() => {
```

Comment:
Running this in an interval instead, so that we keep trying to save until it works (in practice it always seems to work from the first time)

Reply:
The problem was that the `useEffect` didn't re-fire after the nodes were being updated, so it only fired the first time (when the nodes were still empty as the template was still loading)

Reply:
> The problem was that the useEffect didn't re-fire after the nodes were being updated

Do you know why? If `query` is updated, it should re-fire.

Reply:
I think because `query` is a stable reference. I tried adding `query.getSerializedNodes()` as a dependency instead, but it also didn't work because the component itself also didn't update because it had no state change. React's re-render logic is a pain in the ass